jenkins/image_building/dib_elements/ci-base/post-root.d/image_prepull

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+# IMAGE_NAME conflicts with IMAGE_NAME variable in metal3-dev-env, unset it
+unset IMAGE_NAME
+
+# Install container runtime if OS is ubuntu. Centos comes with podman installed
+
+if [[ "${IMAGE_OS}" == "ubuntu" ]]; then
```

Comment:
Please consider putting this block in the install script instead, for consistency.

Reply:
This part is not complete yet, we were facing issue that docker daemon is not run inside the chroot, thus we could not pull images. If you check the commented part, we copied container images folder to environment that is being chrooted. That was a workaround, I will continue when we merge this PR.

Reply:
I think starting  docker service can be put after chroot, but installing part should not.

Another issue I see is that we are mixing the semantics here: we have separate recipes for Ubuntu and Centos, yet we still use `if-else` to differentiate them inside a shared script. Why not put these logics into the OS' respected elements?

Reply:
> I think starting docker service can be put after chroot, but installing part should not.
This all runs on a host environment, container and node images are pulled in a host environment. Then we copy folder containing images to the image that is being built. 
```
sudo cp -a /var/lib/docker/. "${TMP_BUILD_DIR}/mnt/var/lib/docker/"
sudo cp -a /opt/metal3-dev-env/. "${TMP_BUILD_DIR}/mnt/opt/metal3-dev-env/"
```
> Another issue I see is that we are mixing the semantics here: we have separate recipes for Ubuntu and Centos, yet we still use `if-else` to differentiate them inside a shared script. Why not put these logics into the OS' respected elements?

This is incomplete task. We decided to have image building workflow, then continue with improvement tasks.  I will remove it if this is confusing.

Reply:
If this runs on the DIB host, then I'm very tempted to support just one or the other. Is there a reason why we would need to run DIB on both ubuntu and centos? We should be able to build both centos and ubuntu images from an ubuntu host for example.

Reply:
> If this runs on the DIB host, then I'm very tempted to support just one or the other. Is there a reason why we would need to run DIB on both ubuntu and centos? We should be able to build both centos and ubuntu images from an ubuntu host for example.

We do not run DIB in centos. Both centos and ubuntu metal3 images are built in ubuntu host. Current workflow is designed this way.

Reply:
Good! Then we can skip this `if [[ "${IMAGE_OS}" == "ubuntu" ]];` right?